admin/templates/project/streamlit.yaml

```diff
@@ -0,0 +1,82 @@
+#!jinja2: lstrip_blocks: True
+{#
+    REQUIRED APPLICATION VARIABLES
+    ------------------------------
+    image : Docker image path
+    port  : Container port which public HTTP service runs on
+    replicas: how many instances of the application to run
+#}
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: streamlit-app
+  namespace: {{ inventory_hostname }}
+  labels:
+    app: {{ inventory_hostname }}
+spec:
+  replicas: {{ app.replicas }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: {{ inventory_hostname }}
+  template:
+    metadata:
+      labels:
+        app: {{ inventory_hostname }}
+    spec:
+      containers:
+      - image: {{ app.image }}
+        name: {{ inventory_hostname }}
+        ports:
+        - containerPort: {{ app.port }}
+          name: http
+          protocol: TCP
+      imagePullSecrets:
+      - name: regcred
+
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: streamlit-app
```

Comment:
This is copy/paste, I would kind of expect this to be `{{ inventory_hostname }}` if it weren't for symmetry with `public-http` template